outlook/mail/pkg/graph/mail.go

```diff
@@ -37,13 +38,24 @@ func GetMessageDetails(ctx context.Context, client *msgraphsdkgo.GraphServiceCli
 	return result, nil
 }
 
-func SearchMessages(ctx context.Context, client *msgraphsdkgo.GraphServiceClient, subject, fromAddress, fromName, folderID string) ([]models.Messageable, error) {
+func SearchMessages(ctx context.Context, client *msgraphsdkgo.GraphServiceClient, subject, fromAddress, fromName, folderID, start, end string) ([]models.Messageable, error) {
 	var (
 		result models.MessageCollectionResponseable
 		err    error
 		filter []string
 	)
 
+	// It is important that start and end filters are first in the list.
```

Comment:
But the start filter isn't? After reading the linked answer, perhaps
```suggestion
	// It is important that a receivedDateTime filter is first in the list.
```